mqtt/policy/src/core/mod.rs

```diff
@@ -115,8 +115,8 @@ where
                 return match operations.0.get(&request.operation) {
                     // operation exists.
                     Some(resources) => {
-                        // Iterate over and match resources.
-                        // We need to go through all resources and find one with highest priority (smallest order).
+                        // iterate over and match resources.
```

Comment:
nit: a bit complex function with multiple nested loops and a bunch of nested levels. any chance we can improve that?

Reply:
I tried extracting funcs, can't say it looks better that way. That's why I put as much inline comments as I could.